src/MessagePack.AspNetCoreMvcFormatter/MessagePackOutputFormatter.cs

```diff
@@ -41,17 +41,37 @@ public Task WriteAsync(OutputFormatterWriteContext context)
             {
                 if (context.Object == null)
                 {
+#if NETSTANDARD2_0
                     context.HttpContext.Response.Body.WriteByte(MessagePackCode.Nil);
                     return Task.CompletedTask;
+#else
+                    var writer = context.HttpContext.Response.BodyWriter;
+                    var span = writer.GetSpan(1);
+                    span[0] = MessagePackCode.Nil;
+                    writer.Advance(1);
+                    return writer.FlushAsync().AsTask();
+#endif
                 }
                 else
                 {
+#if NETSTANDARD2_0
                     return MessagePackSerializer.SerializeAsync(context.Object.GetType(), context.HttpContext.Response.Body, context.Object, this.options, context.HttpContext.RequestAborted);
+#else
+                    var writer = context.HttpContext.Response.BodyWriter;
+                    MessagePackSerializer.Serialize(context.Object.GetType(), writer, context.Object, this.options, context.HttpContext.RequestAborted);
```

Comment:
Have you confirmed that the `PipeWriter` is optimized for many small `GetSpan` calls as our `Sequence<T>` type is? (i.e. will it allocate a large array, or very many tiny ones?)

Reply:
This is the point I am very interested in.
At least 4K is returned on AspNetCore,
and the standard PipeWriter has a MinimumBufferSize.

I don't know if this can be generalized (as an implicit principle of IBufferWriter) or not.

Reply:
Since you confirmed `PipeWriter` behaves well, I think we can complete the PR.
I don't think `IBufferWriter<T>` is documented as required to create larger than requested buffers, but with `Sequence<T>` and `PipeWriter` doing it well, and likely other buffer writers that come from corefx will too, I think we're in pretty good shape. 
The only thing I think we might improve on is add API documentation where we accept `IBufferWriter<byte>` from the caller that it should be an implementation that allocates large arrays underneath.